faunadb-java-dsl/src/main/java/com/faunadb/client/types/Creators.java

```diff
@@ -0,0 +1,196 @@
+
+            if (method.getParameterTypes().length == 0)
+                return new NoArgsStaticMethodCreator(method);
+
+            return new StaticMethodCreator(method);
+        }
+
+        return null;
+    }
+
+    private static Creator getAnnotatedConstructor(Class<?> clazz) {
+        for (Constructor<?> constructor : clazz.getDeclaredConstructors()) {
+            if (constructor.isAnnotationPresent(FaunaConstructor.class)) {
+                if (constructor.getParameterTypes().length == 0)
+                    return new DefaultConstructorCreator(constructor);
+
+                return new ConstructorCreator(constructor);
+            }
+        }
+
+        return null;
+    }
+
+    private static DefaultConstructorCreator getDefaultConstructor(Class<?> clazz) {
+        try {
+            return new DefaultConstructorCreator(clazz.getDeclaredConstructor());
+        } catch (NoSuchMethodException ex) {
+            return null;
+        }
+    }
+
+    static abstract class AbstractCreator implements Creator {
+        final String[] parameterNames;
+        final SimpleType[] parameterTypes;
+        final Property[] writeProperties;
+
+        AbstractCreator(Constructor<?> constructor) {
+            this.parameterNames = getParameterNames(constructor.getParameterAnnotations());
+            this.parameterTypes = getParameterTypes(constructor.getGenericParameterTypes());
+            this.writeProperties = Properties.getWriteProperties(constructor.getDeclaringClass());
+        }
+
+        AbstractCreator(Method method) {
+            this.parameterNames = getParameterNames(method.getParameterAnnotations());
+            this.parameterTypes = getParameterTypes(method.getGenericParameterTypes());
+            this.writeProperties = Properties.getWriteProperties(method.getDeclaringClass());
+        }
+
+        SimpleType[] getParameterTypes(Type[] genericParameterTypes) {
+            SimpleType[] parameterTypes = new SimpleType[genericParameterTypes.length];
+
+            for (int i = 0; i < genericParameterTypes.length; i++) {
+                parameterTypes[i] = Types.of(genericParameterTypes[i]);
+            }
+
+            return parameterTypes;
+        }
+
+        String[] getParameterNames(Annotation[][] parameterAnnotations) {
+            String[] parameterNames = new String[parameterAnnotations.length];
+
+            for (int i = 0; i < parameterAnnotations.length; i++) {
+                Annotation[] annotations = parameterAnnotations[i];
+
+                for (Annotation annotation : annotations) {
+                    if (annotation.annotationType() == FaunaField.class) {
+                        parameterNames[i] = ((FaunaField) annotation).value();
+                    }
+                }
+
+                if (parameterNames[i] == null)
+                    throw new FaunaException(format("All constructor/creator arguments must be annotated with @%s", FaunaField.class.getSimpleName()));
```

Comment:
It's only possible get the field name using java 8 and change the compilation options to keep the names. Since we're using java 7 currently the only way to get the name is forcing the user annotate on constructors/creators